component.go

```diff
@@ -369,6 +369,12 @@ func (c *Components) UnmarshalBinary(b []byte) error {
 		}
 		b = b[offset+comp.MarshalLen()-2:]
 	}
+
+	c.SetLength()
+	if b[1] != c.Length {
+		return fmt.Errorf("decoded Length is not equal to Components Length, got %d, expected %d", c.Length, b[1])
```

Comment:
I'm not a fan of log-like (redundant) error message - what about this? Or maybe we can even define a custom error like `LengthMismatchError` with some useful fields, but that's perhaps a bit overkill.

```suggestion
		return fmt.Errorf("component length mismatch, got %d, expected %d", c.Length, b[1])
```